### ISSUE TITLE
Remove twisted.test.test_doc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8226]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8226|
|Type|enhancement|
|Created|2016-03-04 07:32:14Z|
|Branch|https://github.com/twisted/twisted/tree/burn-testdoc-8226|

It's trash, we should remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__8226 8226
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_burn_testdoc_8226 branches/burn-testdoc-8226
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1457076734204030 1457076734204030
changetime__1457349887676735 1457349887676735
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46909]) Branching to burn-testdoc-8226.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

It's just dead code in every way.

Builds are all green, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Thanks!

I think that we now have twistedchecker for these kind of tests.

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46941]) Merge burn-testdoc-8226: Remove the useless twisted.test.test_doc

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8226](https://github.com/twisted/twisted/issues/8226)
